app/assets/javascripts/components/stories/History/HistoryChanges.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getDate } from '../../../models/beta/history';
+
+const HistoryChanges = ({ changes }) => {
+  const result = Object.keys(changes).map(key => {
+    const value = getDate(key, changes);
+    return (
+      <div key={key} className="info">
+        <div className="action">
+          {I18n.t(`activerecord.attributes.story.${key}`)}
+        </div>
+        <div className="changes">
+          <div> {value[0] || "---"} </div>
+          <div>
+            <i className="mi md-18">arrow_forward</i>
+          </div>
+          <div> {value[1] || "---"} </div>
+        </div>
+      </div>
+    );
+  });
+
+  return <div> {result} </div>;
```

Comment:
I would create a child component `HistoryChange` and render a list of `HistoryChange`.
Also, you could put `value[0]` and `value[1]` in a variable with a meaningful name.

Reply:
```js
const HistoryChange = ({ title, value0, value1 }) => (
  <div className="info">
    <div className="action">
      {title}
    </div>
    <div className="changes">
      <div>{value0}</div>
      <div>
        <i className="mi md-18">arrow_forward</i>
      </div>
      <div>{value1} </div>
    </div>
  </div>
);

HistoryChange.propTypes = {
  value0: PropTypes.string?, 
  value1: PropTypes.string?,
}

HistoryChange.propTypes = {
  value0: '---', 
  value1: '---',
}

const getTitle = (propName) => I18n.t(`activerecord.attributes.story.${propName}`)

const HistoryChanges = ({ changes }) => 
  mapObject(changes, (val, key) => (
    <HistoryChange title={getTitle(key)} value0={val[0]} value1={val[1]}/>
  ))
```
https://underscorejs.org/#mapObject